sike.go

```diff
@@ -0,0 +1,153 @@
+// This file contains APIs for Supersingular Isogeny Key Encapsulation (SIKE) protocol
+// which is recently submitted to NIST PQC standardization workshop.
+// The underlying core functions are based on SIDH API developed by Cloudflare.
+// SIKE requires NIST's approved hash functions such as sha3 and cSHAKE256 to encapsulate
+// the key based on NIST specifications. 
+//
+//Author: Amir Jalali  				ajalali2016@fau.edu
+//Date: Feb 2018
+
+package p751sidh
+
+import (
+	"io"
+	"golang.org/x/crypto/sha3"
+	"bytes"
+)
+
+const (
+	// The message size, in bytes.
+	MessageSize = 32
+	// The ciphertext size, in bytes = PublicKeySize + MessageSize
+	CiphertextSize = 596
+	// SIKE secret key size, in bytes = MessageSize + PublicKeySize + SecretKeySize
+	SIKESecretKeySize = 644
+	// Custom values for cSHAKE256
+	G = "0"
+	H = "1"
+	P = "2"
+)
+
+type SIKESecretKey struct {
+	Scalar [MessageSize]byte
+	SecretKey *SIDHSecretKeyBob
+	PublicKey *SIDHPublicKeyBob
+}
+
+type SIKECipherText struct {
+	PublicKey *SIDHPublicKeyAlice
+	Scalar [MessageSize]byte
+}
+
+type SIKESharedSecret struct {
+	Scalar [SharedSecretSize]byte
+}
+
+// SIKE keypair generation generates SIKE secret-key and public-key.
+// The secret-key contains a random message + secret-key + public-key.   
+func GenerateKeyPair(rand io.Reader) (publicKey *SIDHPublicKeyBob, SIKEsecretKey *SIKESecretKey, err error) {
```

Comment:
If this function name is "GenerateKeyPair" but the package is p751sidh, then it shares a namespace with GenerateAliceKeyPair and other SIDH-specific functions. I recommend moving this file into its own folder so that it can be an independent module.